lstchain/scripts/lstchain_create_run_summary.py

```diff
@@ -0,0 +1,309 @@
+"""
+Create a run summary for a given date containing the number of subruns,
+the start time of the run, type pf the run: DATA, DRS4, CALI, and
+the reference timestamp and counter of the run.
+"""
+
+import argparse
+import logging
+from collections import Counter
+from datetime import datetime
+from glob import glob
+from pathlib import Path
+
+import numpy as np
+from astropy.table import Table
+from astropy.time import Time
+from ctapipe.containers import EventType
+from ctapipe_io_lst import (
+    CDTS_AFTER_37201_DTYPE,
+    CDTS_BEFORE_37201_DTYPE,
+    DRAGON_COUNTERS_DTYPE,
+    LSTEventSource,
+    MultiFiles,
+)
+from ctapipe_io_lst.event_time import combine_counters
+from traitlets.config import Config
+
+from lstchain import __version__
+from lstchain.paths import parse_r0_filename
+
+log = logging.getLogger(__name__)
+
+parser = argparse.ArgumentParser(description="Create run summary file")
+
+parser.add_argument(
+    "-d",
+    "--date",
+    help="Date for the creation of the run summary in format YYYYMMDD",
+    required=True,
+)
+
+parser.add_argument(
+    "--r0-path",
+    type=Path,
+    help="Path to the R0 files. Default is /fefs/aswg/data/real/R0",
+    default=Path("/fefs/aswg/data/real/R0"),
+)
+
+parser.add_argument(
+    "-o",
+    "--output-dir",
+    type=Path,
+    help="Directory in which Run Summary file is written",
+    default="./",
+)
+
+dtypes = {
+    "ucts_timestamp": np.int64,
+    "run_start": np.int64,
+    "dragon_reference_time": np.int64,
+    "dragon_reference_module_id": np.int16,
+    "dragon_reference_module_index": np.int16,
+    "dragon_reference_counter": np.uint64,
+    "dragon_reference_source": str,
+}
+
+
+def get_list_of_files(r0_path):
+    """
+    Get the list of R0 files from a given date.
+
+    Parameters
+    ----------
+    r0_path : pathlib.Path
+        Path to the R0 files
+
+    Returns
+    -------
+    list_of_files: pathlib.Path.glob
+        List of files
+    """
+    # FIXME: use regular expressions from lstchain.paths.R0_RE
```

Comment:
If you still think this is necessary, it should be as easy as:

```
list_of_files = [f for f in r0_path.glob("LST*.fits.fz") if R0_RE.match(f) is not None]
```

Reply:
Then you would have to list all files in the directory and check the regex on them. I don't really see how that would be better than globbing.

Reply:
I will leave it as it is now then.